deps/rabbit/src/rabbit_stream_queue.erl

```diff
@@ -1039,3 +1041,12 @@ set_leader_pid(Pid, QName) ->
 close_log(undefined) -> ok;
 close_log(Log) ->
     osiris_log:close(Log).
+
+-spec list_with_minimum_quorum() -> [amqqueue:amqqueue()].
+list_with_minimum_quorum() ->
+    lists:filter(fun (Q) ->
+                         StreamId = maps:get(name, amqqueue:get_type_state(Q)),
+                         {ok, Members} = rabbit_stream_coordinator:members(StreamId),
+                         RunningMembers = maps:filter(fun(_, {State, _}) -> State =/= undefined end, Members),
+                         length(maps:values(RunningMembers)) =< length(maps:values(Members)) div 2 + 1
```

Comment:
Despite 10 years or so working with quorum systems I never feel comfortable reading quorum calculations when written in code. :)

In this case would splitting the Members by the predicate above (using `lists:partition/2`) so you have `{Running, NotRunning}` and simply checking that `length(Running) > length(NotRunning)` be sufficient and perhaps easier to read?

Reply:
Either way `map_size` could replace `length(maps:values(` on line 1051

Reply:
I decided to keep the calculation as is, since it's basically the same as in the proven QQ implementation (https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/src/rabbit_quorum_queue.erl#L407-L408).